src/common/string-builder.ts

```diff
@@ -81,8 +82,39 @@ class StringBuilder {
         return this.printQueue.join(' ');
     }
 
-   combineURLs(baseURL: string, relativeURL?: string): string {
-        return relativeURL ? new URL(relativeURL, baseURL || undefined).toString() : baseURL;
+    combineURLs(baseURL: string, relativeURL?: string): string {
+        const createURL = (url: string) => {
+            try {
+                return new URL(url)
+            } catch (e) {
+                return null
+            }
+        }
```

Comment:
There is a mixture of createURL and newURL. I don't think I need createURL if I'm going to use newURL as it is.

Reply:
it's for exception, if you pass path without domain in url - returns null
it is very simple way check relativeUrl contains host or not

Reply:
It can be confusing how to call the createURL function and how to use the new URL right away. People who are not aware of this context do not know why they used the new URL instead of the createURL in the code below. Either you have to change the function name of the createURL a little more intuitively, or use the createURL instead of the new URL in other codes.